syntax = "proto2";

package dharma.common;

enum VehicleBrand {
  RAY = 0;
  LEAPS1 = 1;
}

message VehicleParam {
  optional VehicleBrand brand = 1;
  optional double front_edge_to_center = 2 [default = nan];
  optional double back_edge_to_center = 3 [default = nan];
  optional double left_edge_to_center = 4 [default = nan];
  optional double right_edge_to_center = 5 [default = nan];
    
  optional double length = 6 [default = nan];
  optional double width = 7 [default = nan];
  optional double height = 8 [default = nan];
    
  optional double min_turn_radius = 9 [default = nan];
  optional double max_acceleration = 10 [default = nan];
  optional double max_deceleration = 11 [default = nan];
    
  // The following items are used to compute trajectory constraints in planning/control/canbus,
  // vehicle max steer angle
  optional double max_steer_angle = 12 [default = nan];
  // vehicle max steer rate; how fast can the steering wheel turn.
  optional double max_steer_angle_rate = 13 [default = nan];
  // vehicle min steer rate;
  optional double min_steer_angle_rate = 14 [default = nan];
  // ratio between the turn of steering wheel and the turn of wheels
  optional double steer_ratio = 15 [default = nan];
  // the distance between the front and back wheels
  optional double wheel_base = 16 [default = nan];
  // Tire effective rolling radius (vertical distance between the wheel center
  // and the ground).
  optional double wheel_rolling_radius = 17 [default = nan];
}
